lib/cli-engine.js

```diff
@@ -463,7 +469,23 @@ class CLIEngine {
             });
         }
 
-        this.config = new Config(this.options, this.linter);
+        this.config = new Config(
+            {
+                cwd: this.options.cwd,
+                baseConfig: this.options.baseConfig,
+                rules: this.options.rules,
+                ignore: this.options.ignore,
+                ignorePath: this.options.ignorePath,
+                parser: this.options.parser,
+                parserOptions: this.options.parserOptions,
+                useEslintrc: this.options.useEslintrc,
+                envs: this.options.envs,
+                globals: this.options.globals,
+                configFile: this.options.configFile,
+                plugins: this.options.plugins
+            },
```

Comment:
I changed this to improve encapsulation because it was hard to tell which options were getting used by `Config`. Previously, `Config` would receive all options passed to `CLIEngine`, including irrelevant options like `extensions` and `fix`. With this change, `Config` only receives the options that it actually needs.